shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeSyncServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_CONFIG_MAP_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_KNATIVE_BROKER_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_NETWORK_RESOURCE_READY_NAME;
+import static com.redhat.service.smartevents.infra.v2.api.models.DefaultConditions.DP_SECRET_READY_NAME;
+
+@ApplicationScoped
+public class ManagedBridgeSyncServiceImpl implements ManagedBridgeSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedBridgeSyncServiceImpl.class);
+
+    @Inject
+    ManagerClient managerClient;
+
+    @Inject
+    ManagedBridgeService managedBridgeService;
+
+    @Override
+    public void syncManagedBridgeWithManager() {
+        managerClient.fetchBridgesToDeployOrDelete()
+                .onItem()
+                .invoke(this::processDelta)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully processed ManagedBridge deltas"),
+                        error -> LOGGER.error("Failed to process ManagedBridge deltas", error));
+    }
+
+    @Override
+    public void syncManagedBridgeStatusBackToManager() {
+        managerClient.fetchBridgesToDeployOrDelete()
+                .onItem()
+                .transform(this::transformToBridgeStatus)
+                .invoke(this::notifyBridgeStatus)
+                .subscribe().with(
+                        success -> LOGGER.debug("Successfully sync ManagedBridge status with Manager"),
+                        error -> LOGGER.error("Failed to sync ManagedBridge status with Manager", error));
+    }
+
+    private void processDelta(List<BridgeDTO> bridgeDTOList) {
+        for (BridgeDTO bridgeDTO : bridgeDTOList) {
+            if (bridgeDTO.getOperationType() == OperationType.DELETE) {
+                managedBridgeService.deleteManagedBridge(bridgeDTO);
+            } else {
+                managedBridgeService.createManagedBridge(bridgeDTO);
+            }
+        }
+    }
+
+    private List<BridgeStatusDTO> transformToBridgeStatus(List<BridgeDTO> bridgeDTOList) {
+        Map<String, ManagedBridge> deployedManagedBridges = managedBridgeService.fetchAllManagedBridges()
+                .stream().collect(Collectors.toMap(m -> m.getSpec().getId(), m -> m));
+
+        List<BridgeStatusDTO> bridgeStatusDTOs = new ArrayList<>(bridgeDTOList.size());
+        for (BridgeDTO bridgeDTO : bridgeDTOList) {
+            ManagedBridge deployedManagedBridge = deployedManagedBridges.get(bridgeDTO.getId());
+            if (deployedManagedBridge != null) {
+                bridgeStatusDTOs.add(BridgeStatusConverter.fromManagedBridgeToBrideStatusDTO(deployedManagedBridge));
+            } else {
+                if (bridgeDTO.getOperationType() == OperationType.DELETE) {
```

Comment:
Given the discussion [above](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/1438/files#r1051915340) this code currently serves no purpose and in fact is broken. If you query the `manager` for the list of resources for status updates and then use the `Operation==DELETED` to know whether to send a set of _deleted_ `Condition`'s back you're basically saying "Send me a list of resources; oh, X is deleted I best send you an update saying it has been deleted".

Furthermore, we don't store _deleted_ resources in the database. Once they are deleted they are removed from the tables. The `manager` is therefore not even possible to advise what resources have **actually been deleted**; only those that have been **requested to be deleted**.

Reply:
yes, I mean `OperationType.DELETE` would be **requested to be deleted** so that operator could confirm you back that it has successfully delete the resource.

Reply:
But you're not querying k8s to know if it's been deleted you're querying the `manager` that already knows its been requested to be deleted. The best your code does is convert an `Operation` into a set of `Condition`'s.

Reply:
@vaibhavjainwiz I think all we're missing here is a check to make sure that the CR instance is not in the map of deployed managed bridges.

```
if (bridgeDTO.getOperationType() == OperationType.DELETE) {
  
  / * Add this: make sure that the CR has actually been deleted */
  if(deployedManagedBridge == null) {
      bridgeStatusDTOs.add(getDeletedBridgeStatus(bridgeDTO));
  }
}
```
At the minute you simply assume that if a delete has been requested by the control plane, then it's been deleted by the data plane. The simple `== null` check will verify that the `ManagedBridge` CR, has in fact been deleted.

cc @manstis 

cheers,

Rob

Reply:
@robpblake @manstis 
I think there is a understanding gap, I am querying the k8s to fetch existing CRDs and then check for deleted CRD by comparing with bridge object received from Manager.

here
```
 Map<String, ManagedBridge> deployedManagedBridges = managedBridgeService.fetchAllManagedBridges()
                .stream().collect(Collectors.toMap(m -> m.getSpec().getId(), m -> m));
```

`fetchAllManagedBridges` fetch all deployed bridge CRDs from k8s.

Reply:
With the current implementation you are not sending any update for a resource that is on the shard but (for any reason) not in the fetched resources from the control plane

Reply:
How is it possible for any resource to be on operator but not on control-plane. Even in 1/1000 case it is a case then it make no sense to send a status of resource to control-plane which doesn't exists in there.

Reply:
> @robpblake @manstis I think there is a understanding gap, I am querying the k8s to fetch existing CRDs and then check for deleted CRD by comparing with bridge object received from Manager.
> 
> here
> 
> ```
>  Map<String, ManagedBridge> deployedManagedBridges = managedBridgeService.fetchAllManagedBridges()
>                 .stream().collect(Collectors.toMap(m -> m.getSpec().getId(), m -> m));
> ```
> 
> `fetchAllManagedBridges` fetch all deployed bridge CRDs from k8s.

@vaibhavjainwiz Ah yes, you're absolutely right. Thank you for pointing that out 👍